metricq_tools/energy.py

```diff
@@ -0,0 +1,85 @@
+import asyncio
+import time
+
+import click
+import metricq
+
+from .logging import logger
+from .utils import TemplateStringParam, client_version, metricq_command, run_cmd
+
+
+async def energy(
+    subscriber: metricq.Subscriber, metric: str, command: list[str]
+) -> None:
+    async with subscriber:
+        # No need to keep the subscriber alive, this is just to setup the queue
+        pass
+    time_before = time.time()
+    await run_cmd(command)
+    duration = time.time() - time_before
+    assert duration > 0
+    value_sum = 0.0
+    value_count = 0
+    async with subscriber.drain() as data:
+        async for data_metric, timestamp, value in data:
+            assert data_metric == metric
+            value_sum += value
+            value_count += 1
+
+    click.echo(
+        f"[metricq-energy] duration={duration:.3f} s number of values={value_count}"
+    )
+
+    if value_count == 0:
+        logger.error(f"No data received for {metric}, cannot compute energy.")
+        return
+    if value_count < 10:
+        logger.warning(
+            f"Received only {value_count} data points for {metric}, "
+            f"energy computation may be inaccurate."
+        )
+    value_mean = value_sum / value_count
+    energy = value_mean * duration
+
+    click.echo(f"[metricq-energy] mean of {metric}={value_mean:.1f} W")
```

Comment:
That should at least warrant a new issue: "Actually retrieve and print the unit here".
Obvious use case: `kW` with `kJ`. We may want to push on unit merging first.